cupy/cuda/cub.pyx

```diff
@@ -122,14 +122,28 @@ def device_reduce(ndarray x, op, out=None, bint keepdims=False):
         y = ndarray((kv_bytes,), numpy.int8)
     x_ptr = <void *>x.data.ptr
     y_ptr = <void *>y.data.ptr
-    dtype_id = _get_dtype_id(x.dtype)
+    dtype_id = <int>_get_dtype_id(x.dtype)
     s = <Stream_t>stream.get_current_stream_ptr()
-
-    ws_size = cub_device_reduce_get_workspace_size(x_ptr, y_ptr, x.size, s,
-                                                   op, dtype_id)
+    x_size = <int>x.size
+
+    with nogil:
+        ws_size = cub_device_reduce_get_workspace_size(x_ptr,
+                                                       y_ptr,
+                                                       x_size,
+                                                       s,
+                                                       op,
+                                                       dtype_id)
```

Comment:
You don't have to use `with nogil:` unless the function call takes long time.
(Ditto for other instances)